starter_code/elevator.js

```diff
@@ -3,17 +3,70 @@ class Elevator {
     this.floor      = 0;
     this.MAXFLOOR   = 10;
     this.requests   = [];
+    this.direction = "Up";
+    this.waitingList = [];
+    this.passengers = [];
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start() {
+   setInterval(() => this.update(),  1000);
+	}
+	
+  stop() {
+    clearInterval(setInterval);
+  }
+  update() {
+		if (this.direction === "Up") {
+			this.floorUp();
+		} else {
+			this.floorDown();
+		}
+
+		this.log();
+  }
+  _passengersEnter() {
+		this.waitingList.forEach(function(person, index){
+			if(this.person.originFloor === this.floor) {
+				this.passengers.push(person);
+    		this.waitingList.splice(index, 1);
+    		this.requests.push(person.destinationFloor);
+    		console.log(`${person.name} has entered the elevator`)
+			}
+		})
+  }
+  _passengersLeave() {
+		this.passengers.forEach(function(person, index){
+			if(this.person.destinationFloor === this.floor) {
+    		this.waitingList.splice(index, 1);
+    		console.log(`${person.name} has entered the elevator`)
+			}
+		})
+  }
+  floorUp() {
+    if(this.floor !== this.MAXFLOOR) {
+			this.floor++;
+			this.waitingList.filter(this.floor);
```

Comment:
Remember to look at how filter works; you need to give it a "test" so that it will filter out the things don't pass.